app/src/main/cpp/skyline/services/am/storage/IStorageAccessor.cpp

```diff
@@ -8,34 +8,37 @@ namespace skyline::service::am {
     IStorageAccessor::IStorageAccessor(const DeviceState &state, ServiceManager &manager, std::shared_ptr<IStorage> parent) : parent(std::move(parent)), BaseService(state, manager) {}
 
     Result IStorageAccessor::GetSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<i64>(static_cast<i64>(parent->content.size()));
+        response.Push<i64>(static_cast<i64>(parent->GetSpan().size()));
         return {};
     }
 
     Result IStorageAccessor::Write(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         auto offset{request.Pop<i64>()};
 
-        if (offset < 0 || offset > parent->content.size())
+        if(!parent->writable)
+            return result::ObjectInvalid;
+
+        if (offset < 0 || offset > parent->GetSpan().size())
             return result::OutOfBounds;
 
-        size_t size{std::min(request.inputBuf.at(0).size(), parent->content.size() - static_cast<size_t>(offset))};
+        size_t size{std::min(request.inputBuf.at(0).size(), parent->GetSpan().size() - static_cast<size_t>(offset))};
```

Comment:
Same as before, make the span a local variable rather than using it directly in-line, it looks very unneat.